daemon/observer/observer.go

```diff
@@ -132,13 +135,17 @@ func (n *Notifier) Notify(eventType EventType, message string, increment bool) {
 	}
 
 	if increment {
+		n.Lock()
 		n.current++
+		n.Unlock()
 	}
 
+	n.RLock()
 	if n.current > n.total {
 		panic(fmt.Sprintf(
```

Comment:
I have changed this as this could indeed cause locking issues seeing that the panic is being recovered higher up the chain.

It now reads the data into two variables, isolated in a single rlock/runlock.